classes/local/source/cohort.php

```diff
@@ -300,4 +299,27 @@ public static function fix_allocations(?int $programid, ?int $userid): bool {
 
         return $updated;
     }
+
```

Comment:
Just above this line, there is a comment:

// NOTE: it is expected that enrolment fixing is executed right after this method.

This is no longer true, is this going to be an issue?

Reply:
I hadn't changed any of spots where `allocation::fix_allocation_sources` is called, so they are all still followed by `allocation::fix_user_enrolments`. So there should be no issue with that.

Reply:
I see that there is enrolments added to the allocation chunk, so this might be fine

Reply:
![image](https://github.com/catalyst/moodle-enrol_programs/assets/21162968/0cdb9926-5b1e-41dc-8a61-408bd3e2e769)

Its moved here in cron.php @rhell4

Reply:
The comment in question is alluding to the user enrolments, which ARE still after the allocation sources. Resolving this one